src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/GetSetTimes_SafeFileHandle.cs

```diff
@@ -96,19 +96,23 @@ protected override DateTime GetLastWriteTimeUtc(string path)
         public async Task WritingShouldUpdateWriteTime_After_SetLastAccessTime()
         {
             string filePath = GetTestFilePath();
-            using var handle = OpenFileHandle(filePath, FileAccess.ReadWrite);
+            using SafeFileHandle handle = OpenFileHandle(filePath, FileAccess.ReadWrite);
 
             File.SetLastAccessTime(handle, DateTime.Now.Subtract(TimeSpan.FromDays(1)));
-            var timeBeforeWrite = File.GetLastWriteTime(handle);
+            DateTime timeAfterWrite = default, timeBeforeWrite = File.GetLastWriteTime(handle);
 
-            using var writer = new StreamWriter(new FileStream(handle, FileAccess.ReadWrite));
-            writer.AutoFlush = true;
-            writer.WriteLine("now: " + DateTime.Now);
-            await Task.Delay(2000);
-            writer.WriteLine("now: " + DateTime.Now);
+            // According to https://learn.microsoft.com/en-us/windows/win32/api/fileapi/ns-fileapi-win32_file_attribute_data
+            // write time has a resolution of 2 seconds on FAT. Let's wait a little bit longer.
```

Comment:
Is the problem manifesting only on FAT? I see x64 builds also failing and I thought those used NTFS.

Reply:
No, it's not. Afair we don't run tests on FAT.

My point here was to show that time resolution exists and we need at least 2 seconds because of that.

Reply:
Yeah the 2 seconds delay was chosen after reading that page. I forgot to mention it in the testcase